### ISSUE TITLE
Crash when chronotank tries to teleport

### ISSUE THREAD

**Issue by Dasha-98** [OpenRA/OpenRA#20446]
## Issue Summary

When a chronotank tries to teleport, the game crashes.

## Reproduction

- Start a multiplayer game or skirmish with Germany.
- Create a tech center in order to be able to build a chronotank.
- Create a chronotank.
- Try to teleport it.
- The game will crash.

## Expected behavior

Smooth multiplayer game without crash.

## Screenshots / Screen recordings / Replays

Here's the generated log file exception-2022-11-07T220103Z.log : 

OpenRA engine version release-20210321
Red Alert mod version release-20210321
on map b90e6ff65786aa93dd6f4ce07280801a6ad60731 (Italy by Westwood Studios).
Date: 2022-11-07 22:01:03Z
Operating System: Linux (Unix 4.15.0.136)
Runtime Version: Mono 6.4.0.198 (tarball Tue Sep 24 01:20:30 UTC 2019) CLR 4.0.30319.42000
Exception of type `System.IO.InvalidDataException`: chrotnk1.aud is not a valid sound file!
  at OpenRA.Sound.LoadSound[T] (System.String filename, System.Func`2[T,TResult] loadFormat) [0x0009e] in <24adfa0d4e934ea8abdc965bd2260629>:0 
  at OpenRA.Sound+<>c__DisplayClass15_0.<Initialize>b__1 (System.String filename) [0x00000] in <24adfa0d4e934ea8abdc965bd2260629>:0 
  at OpenRA.Exts.GetOrAdd[K,V] (System.Collections.Generic.Dictionary`2[TKey,TValue] d, K k, System.Func`2[T,TResult] createFn) [0x0000b] in <24adfa0d4e934ea8abdc965bd2260629>:0 
  at OpenRA.Primitives.Cache`2[T,U].get_Item (T key) [0x00000] in <24adfa0d4e934ea8abdc965bd2260629>:0 
  at OpenRA.Sound.Play (OpenRA.SoundType type, OpenRA.Player player, System.String name, System.Boolean headRelative, OpenRA.WPos pos, System.Single volumeModifier, System.Boolean loop) [0x00030] in <24adfa0d4e934ea8abdc965bd2260629>:0 
  at OpenRA.Sound.Play (OpenRA.SoundType type, System.String name, OpenRA.WPos pos) [0x00000] in <24adfa0d4e934ea8abdc965bd2260629>:0 
  at OpenRA.Mods.Cnc.Activities.Teleport.Tick (OpenRA.Actor self) [0x00091] in <dcef2b84f6e940d2829acb63eee7e11c>:0 
  at OpenRA.Activities.Activity.TickOuter (OpenRA.Actor self) [0x00084] in <24adfa0d4e934ea8abdc965bd2260629>:0 
  at OpenRA.Traits.ActivityUtils.RunActivity (OpenRA.Actor self, OpenRA.Activities.Activity act) [0x00015] in <24adfa0d4e934ea8abdc965bd2260629>:0 
  at OpenRA.Actor.Tick () [0x0000f] in <24adfa0d4e934ea8abdc965bd2260629>:0 
  at OpenRA.World.Tick () [0x00144] in <24adfa0d4e934ea8abdc965bd2260629>:0 
  at OpenRA.Game.InnerLogicTick (OpenRA.Network.OrderManager orderManager) [0x001bc] in <24adfa0d4e934ea8abdc965bd2260629>:0 
  at OpenRA.Game.LogicTick () [0x0003e] in <24adfa0d4e934ea8abdc965bd2260629>:0 
  at OpenRA.Game.Loop () [0x000f1] in <24adfa0d4e934ea8abdc965bd2260629>:0 
  at OpenRA.Game.Run () [0x0003c] in <24adfa0d4e934ea8abdc965bd2260629>:0 
  at OpenRA.Game.InitializeAndRun (System.String[] args) [0x00010] in <24adfa0d4e934ea8abdc965bd2260629>:0 
  at OpenRA.Launcher.Program.Main (System.String[] args) [0x00044] in <654d601744ea40a08007c756fb7dd724>:0

**Comment by PunkPun** [comment]
looks like something wrong with chronotank. Are you sure the map wasn't modded?

`chrotnk1.aud is not a valid sound file`

**Comment by Dasha-98** [comment]
It's a map I downloaded sometime ago on the Resource Center.
This is the one : https://resource.openra.net/maps/13684/ 
(checksums are matching)

**Comment by abcdefg30** [comment]
This looks like an old issue with the assets installation (CNCNZ ISOs), see https://github.com/OpenRA/OpenRA/issues/12772#issuecomment-279390848 and https://github.com/OpenRA/OpenRA/issues/12420#issuecomment-262778529.

**Comment by Dasha-98** [comment]
I could just reproduce it on other maps in Skirmish mode.
As soon as I try to teleport a chronotank, the crash happens.

I'm attaching the logs and replays here.

[ra-2022-11-10T095339Z.zip](https://github.com/OpenRA/OpenRA/files/9979657/ra-2022-11-10T095339Z.zip)

[ra-2022-11-10T100105Z.zip](https://github.com/OpenRA/OpenRA/files/9979659/ra-2022-11-10T100105Z.zip)

[exception-2022-11-10T100008Z.log](https://github.com/OpenRA/OpenRA/files/9979661/exception-2022-11-10T100008Z.log)

[exception-2022-11-10T100616Z.log](https://github.com/OpenRA/OpenRA/files/9979664/exception-2022-11-10T100616Z.log)

I've updated the issue description.

**Comment by abcdefg30** [comment]
Yes, it is due to a corrupted assets installation, see the comments linked above. To fix this you'd have to delete content again and re-download it.

**Comment by Dasha-98** [comment]
Let me try it and report back.

**Comment by Dasha-98** [comment]
I wiped out the content and did a "quick install" (without music or anything).
Now it's fine. I'll try to re-install the full content and retest.

Note: if one wants to investigate or test this issue, I've created a save game right after the Chrono Tank has been produced.

[Test_Chrono_Tank.zip](https://github.com/OpenRA/OpenRA/files/9980170/Test_Chrono_Tank.zip)

**Comment by Dasha-98** [comment]
I've installed the allied (disc 1) and soviet (disc 2) and it still works.

Now, I'll see how I can install the counterstrike (disc 3) and (disc 4) as my dumps are in .cue + .bin (multiple tracks) and I don't have an easy way to mount a .cue file (so that all audio tracks are properly loaded) in Ubuntu.

**Comment by Dasha-98** [comment]
Update : I used CDEmu to properly mount the .cue + .bin image of my Counterstrike and Aftermath CD's.

And when the image is properly mounted, the music from Counterstrike and Aftermath can be installed and the game does not crash anymore.

I think I know what is the issue : 
- The previous image of Counterstrike and Afttermath I had were in ISO format.
- But the expansion discs are in fact mixed CD (music + data). And ISO is not a good way to handle these mixed CD's (only the data part will be retained in the ISO). For these CDs, it's better to use .cue and .bin or .ccd/.cue and .img.

So when I used the ISO (for convenience) to install the music, it could not be copied properly and this is what created this issue probably.

Having the image properly mounted (with CDEmu) as a mixed disc solves the issue.

One point which is sad is that CD's from other languages cannot be used. I also own the French discs but they are not recognized by the installer (neither the base game nor the expansions).
